kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -103,13 +104,27 @@ public static String namespace(@NonNull HasMetadata resource) {
     }
 
     public static Long generation(@NonNull HasMetadata resource) {
-        return resource.getMetadata().getGeneration();
+        return getGeneration(resource);
     }
 
     public static String uid(@NonNull HasMetadata resource) {
         return resource.getMetadata().getUid();
     }
 
+    /**
+     * Extract generation from a MetaData object.
+     *
+     * @param observedGenerationSource the object from which to extract the metadata generation.
+     * @return the metadata generation of <code>0</code> of te metadata or the generation itself is null in alignment with @see <a href="https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions#kep-1623-standardize-conditions">KEP 1623</a>.
+     */
+    public static @NonNull Long getGeneration(HasMetadata observedGenerationSource) {
```

Comment:
`generation` and `getGeneration` are redundant. Maybe inline `getGeneration` since we already have other methods like `name`, `uid` etc. Or switch them all to `get*`.

I also think we don't need to take about `observedGeneration` in here, we're getting the generation of a resource,  the client code will be the place that knows it's going to be used as an `observedGeneration`.

Reply:
also not sure that we need to guard `null` metadata, I don't think the Operator will be able to encounter null metadata, resources have to have name/namespace from the moment they are born right?.